inequality/pen.py

```diff
@@ -0,0 +1,337 @@
+"""
+Pen's Parade and Pengram Visualizations
+
+This module provides functions to create Pen's Parade visualizations and
+extend them with choropleth maps to display the spatial distribution of
+values. The `pen` function generates a traditional Pen's Parade, which is
+a visual representation of income distribution or similar data, typically
+used to show inequality. The `pengram` function enhances this by combining
+the Pen's Parade with a choropleth map, allowing for a richer analysis of
+spatial data distributions.
+
+Author
+------
+Serge Rey <srey@sdsu.edu>
+"""
+
+import math
+
+import numpy as np
+
+
+def pen(
+    df,
+    col,
+    x,
+    weight=None,
+    ascending=True,
+    xticks=True,
+    total_bars=100,
+    figsize=(8, 6),
+    ax=None,
+):
+    """
+    Creates the Pen's Parade visualization.
+
+    This function generates a bar plot sorted by a specified column, with
+    options to customize the x-axis ticks and figure size. The Pen's Parade
+    is a visual representation of income distribution (or similar data),
+    typically used to show inequality.
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        DataFrame containing the data to plot.
+    col : str
+        The column to plot on the y-axis.
+    x : str
+        The column to plot on the x-axis.
+    weight : str, optional
+        A column used to weight the bars in the Pen’s Parade. Default is None.
+    ascending : bool, optional
+        Whether to sort the DataFrame in ascending order by the `col`.
+        Default is True.
+    xticks : bool, optional
+        Whether to show x-axis ticks. Default is True.
+    total_bars : int, optional
+        Total number of bars to create for the weighted Pen’s Parade. Default
+        is 100.
+    figsize : list, optional
+        The size of the figure as a list [width, height]. Default is [8, 6].
+    ax : matplotlib.axes.Axes, optional
+        Matplotlib Axes instance to plot on. If None, a new figure and axes
+        will be created. Default is None.
+
+    Returns
+    -------
+    matplotlib.axes.Axes
+        A Matplotlib Axes object with the Pen's Parade plot.
+
+    """
+    try:
+        import seaborn as sns
+    except ImportError as e:
+        raise ImportError(
+            "Seaborn is required for pen. Install with  'pip install seaborn'."
```

Comment:
```suggestion
            "Seaborn is required for pen. Install with 'pip install seaborn'."
```

Also, is there a reason for suggested `pip` over `conda` here and below?